components/app_bar/app_bar.tsx

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {CSSProperties} from 'react';
+import classNames from 'classnames';
+import semver from 'semver';
+
+import {AppBinding} from 'mattermost-redux/types/apps';
+import {Channel} from 'mattermost-redux/types/channels';
+import {MarketplaceApp, MarketplacePlugin} from 'mattermost-redux/types/marketplace';
+import {Theme} from 'mattermost-redux/types/themes';
+
+import {PluginComponent} from 'types/store/plugins';
+
+import AppBarBinding from './app_bar_binding';
+
+import './app_bar.scss';
+
+export type Props = {
+    channelHeaderComponents: PluginComponent[];
+    appBarBindings: AppBinding[];
+    theme: Theme;
+    channel: Channel;
+    marketplaceListing: Array<MarketplacePlugin | MarketplaceApp>;
+    activePluginId?: string;
+
+    actions: {
+        fetchListing: () => Promise<{ data?: Array<MarketplacePlugin | MarketplaceApp>}>;
+    };
+}
+
+type State = {
+    loadedMarketplaceListing: boolean;
+}
+
+export default class AppBar extends React.PureComponent<Props> {
```

Comment:
just curious: why is this not written as a functional component?

Reply:
Converted to a function component